client/src/views/Simulation/Simulation.vue

```diff
@@ -145,43 +155,68 @@
      *  needs to trigger a refresh of the results. */
     get triggerFetchResults (): string {
       const { end, start, step } = this.runConfig;
+      const currentRunAllParametersValues = [];
+      this.getSelectedModelIds.map(modelId => {
+        const modelParameters = this.getSimParameterArray(Number(modelId));
+        // Just check the current run change of values
+        if (modelParameters[0]) {
+          currentRunAllParametersValues.push(...Object.values(modelParameters[0]));
+        }
+      });
       const watchObject = {
         config: { end, start, step },
-        parameters: [...this.getSimParameterArray],
+        parameters: currentRunAllParametersValues.filter(Boolean),
       };
       return JSON.stringify(watchObject);
     }
 
-    get selectedModel (): Model.Model {
+    get selectedModels (): Model.Model[] {
       if (
-        !this.getSelectedModelIds?.[0]?.toString() && // Are we missing the selectedModelId, toString to test id 0 as well,
+        this.getSelectedModelIds.length < 1 && // Are we missing the selectedModelId, toString to test id 0 as well,
         this.$route.params.model_id && // Does the model id is available from the route parameters,
-        !Number.isNaN(this.$route.params.model_id) // Make sure the model id from the route is not a NaN.
+        typeof this.$route.params.model_id === 'string' // Make sure the model id from the route is a string.
       ) {
         // Set the model id from the route as the selected model.
-        this.setSelectedModels(Number(this.$route.params.model_id));
+        this.$route.params.model_id.split(',').forEach(this.setSelectedModels);
       }
-      return this.getModelsList.find(model => model.id === Number(this.getSelectedModelIds[0]));
+      return this.getModelsList.filter(model => this.getSelectedModelIds.map(Number).includes(model.id));
     }
 
     get gridMap (): string[][] {
-      return this.expandedId
-      ? [[this.expandedId]]
-      : [
-          ['model', 'div1', 'parameters', 'div2', 'variables'],
-        ];
+      const gridMap: string[][] = [];
+
+      this.selectedModels.forEach((model, index) => {
+        if (this.expandedId) {
+          gridMap.push([this.expandedId + '_' + index]);
+        } else if (this.setSelectedModels) {
+          gridMap.push([
+            'model_' + index,
+            'model-parameters-separator',
+            'parameters_' + index,
+            'parameters-variables-separator',
+            'variables_' + index,
+          ]);
+        }
+        // gridMap.push(['row-separator']);
```

Comment:
Remove commented code.

Reply:
This is still a draft, and there is a lot of bugs to iron out, this being one of them.